src/js/Router.js

```diff
@@ -64,22 +78,26 @@ export default class Router extends String {
      * route().current('posts.show', { post: 1 }); // false
      * route().current('posts.show', { post: 4 }); // true
      *
-     * @param {String} name - Route name to check.
+     * @param {String|Boolean} name - Route name to check.
      * @param {(String|Number|Array|Object)} params - Route parameters.
-     * @return {(Boolean|String|undefined)}
+     * @return {(Boolean|String|Array|undefined)}
      */
-    current(name, params) {
-        const url = this._config.absolute
-            ? this._location().host + this._location().pathname
-            : this._location().pathname.replace(this._config.url.replace(/^\w*:\/\/[^/]+/, ''), '').replace(/^\/+/, '/');
-
+    current(name = '', params = undefined) {
+        const url = this._currentUrl();
         // Find the first route that matches the current URL
+        let matchedParams = null;
         const [current, route] = Object.entries(this._config.routes).find(
-            ([_, route]) => new Route(name, route, this._config).matchesUrl(url)
+            ([_, route]) => (matchedParams = new Route(name, route, this._config).matchesUrl(url))
         ) || [undefined, undefined];
 
+        const {_query: query, ...rParams} = matchedParams;
+        matchedParams = {...rParams, ...query};
+        // TODO params should probably return the query params as _query, same way they are passed to compile, but would be a breaking change
```

Comment:
Would this accomplish the same thing?

```suggestion
        matchedParams = { ...matchedParams.rParams, ...matchedParams._query };
```

Reply:
Ah right okay, thanks!